t/03-testapi.t

```diff
@@ -345,7 +345,12 @@ subtest 'script_run' => sub {
     $fake_exit = 1234;
     is(background_script_run('sleep 10'), '1234', 'background_script_run returns a PID');
     is(background_script_run('sleep 10', output => 'foo'), '1234', 'background_script_run with output returns valid PID');
-};
+
+    my $mock_testapi = Test::MockModule->new('testapi');
+    $mock_testapi->redefine(is_serial_terminal => 1);
+    $mock_testapi->redefine(hashed_string => 'XXX');
+    stderr_like { is(assert_script_run('true'), undef, 'nothing happens on success') } qr/wait_serial: (?^u:XXX-\d+-): ok/, 'log';
```

Comment:
It would be nice if the test messages (`nothing happens on success` and `log`) would be unique within one test-level (here, this sub test). Maybe you can just merge this with the place you've copied these from (as the original test is quite unspecific anyways)?

Reply:
This part does not have coverage, will try to do so.
![Screenshot_2023-04-04_15-05-14](https://user-images.githubusercontent.com/9447312/229807485-74f35e83-2f4b-4030-a057-f90a02dffe4c.png)

Reply:
@Martchus I'm not able to add working unit test now, I will do it later. PR can stay open, it's not super important.